flink-core/src/main/java/org/apache/flink/util/NewSplittableIterator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import org.apache.flink.annotation.Public;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Abstract base class for iterators that can split themselves into multiple disjoint iterators. The
+ * union of these iterators returns the original iterator values.
+ *
+ * @param <T> The type of elements returned by the iterator.
+ */
+@Public
+public abstract class NewSplittableIterator<T> implements Iterator<T>, Serializable {
```

Comment:
The name is not perfect ;)

Reply:
This is just to discuss deprecation, not intended to be called like that.

Reply:
Is the only proposed change to adjust the signature of `split` from arrays to lists? Can you give a quick explanation about the motivation?